### PR TITLE
Support HTTP based Azure SDK service tracing by Spring Sleuth implementation

### PR DESCRIPTION
**Pull request by moarychan**
You can find the [feature proposal here](https://github.com/Azure/azure-sdk-for-java/issues/21969), this PR will support the [service HTTP based](https://github.com/Azure/azure-sdk-for-java/issues/24651), the tracer is delegated to Spring Sleuth, it's available to record the service activities when enabling the Spring Sleuth function.

### CODE REVIEW

File: sdk/spring/azure-spring-cloud-trace-sleuth/src/main/java/com/azure/spring/tracing/sleuth/SleuthHttpPolicy.java [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    public int getOrder() {
+        return Ordered.LOWEST_PRECEDENCE;
+    }
+
+    // Singleton Sleuth tracer capable of starting and exporting spans.
+    private final Tracer tracer;
+    private final Propagator propagator;
+
+    // standard attributes with http call information
+    private static final String HTTP_USER_AGENT = "http.user_agent";
```
[the remark this answers is not in the document]
Comment (stliu) [reply sdk/spring/azure-spring-cloud-trace-sleuth/src/main/java/com/azure/spring/tracing/sleuth/SleuthHttpPolicy.java:48]: also, which stage will this policy be added into pipeline, before or after retry? will retry requests be tracked?

File: sdk/spring/azure-spring-cloud-trace-sleuth/src/main/java/com/azure/spring/tracing/sleuth/implementation/HttpTraceUtil.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.tracing.sleuth.implementation;
+
+import org.springframework.cloud.sleuth.Span;
+
+public final class HttpTraceUtil {
+
+    private static final String STATUS_100 = "Continue";
```
Comment (stliu) [inline sdk/spring/azure-spring-cloud-trace-sleuth/src/main/java/com/azure/spring/tracing/sleuth/implementation/HttpTraceUtil.java:10]: I don't think you need to define these stuff by youslef, how about using something existing, like `org.springframework.http.HttpStatus`

File: sdk/spring/azure-spring-cloud-trace-sleuth/src/main/java/com/azure/spring/tracing/sleuth/implementation/SpanId.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+ *
+ * <p>There are two more other representation that this class helps with:
+ *
+ * <ul>
+ *   <li>Bytes: a 8-byte array, where valid means that at least one of the bytes is not `\0`.
+ *   <li>Long: a {@code long} value, where valid means that the value is non-zero.
+ * </ul>
+ */
+@Immutable
+public final class SpanId {
```
Comment (stliu) [inline sdk/spring/azure-spring-cloud-trace-sleuth/src/main/java/com/azure/spring/tracing/sleuth/implementation/SpanId.java:22]: this is no such class within spring-cloud-sleuth project?

File: sdk/spring/azure-spring-cloud-trace-sleuth/src/main/java/com/azure/spring/tracing/sleuth/SleuthHttpPolicy.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (moarychan) [reply sdk/spring/azure-spring-cloud-trace-sleuth/src/main/java/com/azure/spring/tracing/sleuth/SleuthHttpPolicy.java:48]: this policy will be positioned HttpPipelinePosition.PER_RETRY by default, it uses the default pipeline position, https://github.com/Azure/azure-sdk-for-java/blob/ed7d8935596a83c81d2552e4fd8e97a9cd471bff/sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpPipelinePolicy.java#L37; the retry request will be tracked.

File: sdk/spring/azure-spring-cloud-trace-sleuth/src/main/java/com/azure/spring/tracing/sleuth/implementation/SpanId.java (hunk 3 above)
Comment (moarychan) [reply sdk/spring/azure-spring-cloud-trace-sleuth/src/main/java/com/azure/spring/tracing/sleuth/implementation/SpanId.java:22]: Yes, not found a suitable class.

File: sdk/spring/azure-spring-cloud-trace-sleuth/src/main/java/com/azure/spring/tracing/sleuth/implementation/HttpTraceUtil.java (hunk 2 above)
Comment (moarychan) [reply sdk/spring/azure-spring-cloud-trace-sleuth/src/main/java/com/azure/spring/tracing/sleuth/implementation/HttpTraceUtil.java:10]: OK, the dependency `org.springframework:spring-web` will be added.

**Review by saragluna — APPROVED** [review]

**Pull request merged**